process/block/block_processor.go

```diff
@@ -1,18 +1,123 @@
+	"github.com/ElrondNetwork/covalent-indexer-go/process/utility"
 	"github.com/ElrondNetwork/covalent-indexer-go/schema"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
 	"github.com/ElrondNetwork/elrond-go-core/data"
+	erdBlock "github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go-core/data/indexer"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
 )
 
-type blockProcessor struct{}
+const DefaultProposerIndex = int64(0)
+
+type blockProcessor struct {
+	marshaller        marshal.Marshalizer
+	miniBlocksHandler process.MiniBlockHandler
+}
 
 // NewBlockProcessor creates a new instance of block processor
-func NewBlockProcessor() (*blockProcessor, error) {
-	return &blockProcessor{}, nil
+func NewBlockProcessor(marshaller marshal.Marshalizer, mbHandler process.MiniBlockHandler) (*blockProcessor, error) {
+	if check.IfNil(marshaller) {
+		return nil, covalent.ErrNilMarshaller
+	}
+	if mbHandler == nil {
+		return nil, covalent.ErrNilMiniBlockHandler
+	}
+
+	return &blockProcessor{
+		marshaller:        marshaller,
+		miniBlocksHandler: mbHandler,
+	}, nil
 }
 
 // ProcessBlock converts block data to a specific structure defined by avro schema
-func (bp *blockProcessor) ProcessBlock(block data.BodyHandler) (*schema.Block, error) {
-	return nil, nil
+func (bp *blockProcessor) ProcessBlock(args *indexer.ArgsSaveBlockData) (*schema.Block, error) {
+	blockSizeInBytes, err := bp.computeBlockSize(args.Header, args.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	miniBlocks, err := bp.miniBlocksHandler.ProcessMiniBlocks(args.Header, args.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	header := args.Header
+	return &schema.Block{
+		Nonce:                 int64(header.GetNonce()),
+		Round:                 int64(header.GetRound()),
+		Epoch:                 int32(header.GetEpoch()),
+		Hash:                  args.HeaderHash,
+		MiniBlocks:            miniBlocks,
+		NotarizedBlocksHashes: utility.StrSliceToBytesSlice(args.NotarizedHeadersHashes),
+		Proposer:              getProposerIndex(args.SignersIndexes),
+		Validators:            utility.UIntSliceToIntSlice(args.SignersIndexes),
+		PubKeysBitmap:         header.GetPubKeysBitmap(),
+		Size:                  blockSizeInBytes,
+		Timestamp:             int64(header.GetTimeStamp()),
+		StateRootHash:         header.GetRootHash(),
+		PrevHash:              header.GetPrevHash(),
+		ShardID:               int32(header.GetShardID()),
+		TxCount:               int32(header.GetTxCount()),
+		AccumulatedFees:       utility.GetBytes(header.GetAccumulatedFees()),
+		DeveloperFees:         utility.GetBytes(header.GetDeveloperFees()),
+		EpochStartBlock:       header.IsStartOfEpochBlock(),
+		EpochStartInfo:        getEpochStartInfo(header),
+	}, nil
+}
+
+func (bp *blockProcessor) computeBlockSize(header data.HeaderHandler, body data.BodyHandler) (int64, error) {
```

Comment:
block size on the indexer is calculated also with transactions.
If you do this will have different values.

Reply:
This is the same implementation as in indexer: 
![image](https://user-images.githubusercontent.com/82832880/130816761-87c5f97f-9bfd-4b43-85cc-354fcd097259.png)

Txs size is a different field in indexer, whereas here it does not exist